pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -175,4 +183,41 @@ public String getCrcMetadataForTable(
       }
     }
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  @Path("/tables/{tableName}/segments/{segmentName}")
+  @ApiOperation(value = "Download a segment", notes = "Download a segment in zipped tar format")
+  public Response downloadSegment(
+      @ApiParam(value = "Name of the table with type REALTIME OR OFFLINE", required = true, example = "myTable_OFFLINE") @PathParam("tableName") String tableName,
+      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
+      @Context HttpHeaders httpHeaders)
+      throws Exception {
+    LOGGER.info("Get a request to download segment {} for table {}", segmentName, tableName);
+    TableDataManager tableDataManager = checkGetTableDataManager(tableName);
+    SegmentDataManager segmentDataManager = tableDataManager.acquireSegment(segmentName);
+    if (segmentDataManager == null) {
+      throw new WebApplicationException(String.format("Table %s segments %s does not exist", tableName, segmentName),
+          Response.Status.NOT_FOUND);
+    }
+    try {
```

Comment:
Until now, the only way to download a segment was via controller API. With this, we have introduced a new API to download segment. So, we need to add the access control here. See `PinotSegmentUploadDownloadRestletResource`

The general approach we have take is to have different AccessControl interface and factory declarations for each component, so you should add one for the server. 

I am ok if it is done in a different PR, just mark as a TODO here. and include another independent PR

Reply:
Done.

Reply:
@mcvsubbu Are we going to merge all these PRs together? We cannot deploy a version without access control on segment download

Reply:
@Jackie-Jiang why do you think we cannot deploy a version without access control?

Reply:
@mcvsubbu Because we are GDPR compliance. I don't think we should allow downloading without access control.

Reply:
@Jackie-Jiang Pinot open source is not GDPR compliant. You can see that even though Controller APIs have a call out to Auth layer, the default is  allow-all authenticator. The same holds for the server as well.

It is up to the admins to decide how to plug-in  appropriate auth systems. I am expecting that @chenboat  will add an auth layer callout either in this or immediately following PR

Reply:
@mcvsubbu @Jackie-Jiang I prefer to adding the auth layer in a PR immediately following this one. Looking at PinotSegmentUploadDownloadRestletResource, it seems that we can reuse the same AccessControlFactory and AccessControl class, right?

Reply:
Like I said before, I believe we named them the same, but they exist as different classes, one in each "product". So, you need to create one for server.  You cannot create a dependency from server onto controller

Reply:
I was thinking about moving AccessControlFactory and AccessControl classes to Pinot-common or core to reuse the classes. This will also avoid introduce server to controller dependency. Is any issue with this approach?

Reply:
These may not be common, and may evolve to be different classes. The kind of auth that one may want fo server may be very different from that of what is needed in Broker vs Controller. It is best to keep them apart.

Reply:
I will add a separate AccessControlFactory class in an immediate follow-on PR. The reason is to keep this PR's focus. @Jackie-Jiang, please let me know if you still have issue with this plan.

Reply:
@chenboat I'm okay addressing it later